velox/exec/Driver.cpp

```diff
@@ -68,13 +68,16 @@ class CancelGuard {
 // input channel if so.
 std::optional<column_index_t> getIdentityProjection(
     const std::vector<IdentityProjection>& projections,
-    column_index_t outputChannel) {
+    column_index_t outputChannel,
+    Operator* prevOp) {
   for (const auto& projection : projections) {
     if (projection.outputChannel == outputChannel) {
       return projection.inputChannel;
     }
   }
-  return std::nullopt;
+
+  // If grouping key and join key is the same, continue walking upstream.
+  return prevOp->getGroupKeyInputChannel(outputChannel);
```

Comment:
This projection is no longer identity.  We don't need identity for filter pushdown to pass through, we only need the projection to be a subset of the its input.  Can we add `Operator::filteredProjections()` and use that for filter pushdown instead?  For most operators this should be forward to `identityProjections()` and for `HashAggregation` you can add all keys.

Reply:
That's a good suggestion.